.github/actions/publish-nuget/action.yml

```diff
@@ -60,7 +60,13 @@ runs:
         ${{ inputs.publish-version }}
         '@.Trim()
 
-        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_REF_NAME') {
+            $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
+            if (!($version -match '^\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
+                Write-Output "Invalid version number! The correct format is vM.N.O or vM.N.O-alpha.X.[issue-code]."
```

Comment:
Please make it a [workflow command](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message) (so it generates an error annotation) by adding `::error::` at the beginning of the string.

Also I'd include the current value of `$version` in the message so the problem is very clear on the summary page too.